### PR TITLE
Feature/show different days in debates

### PR DESCRIPTION
**Pull request by cf17jeremy**
#### :tophat: What? Why?
*Adds multiple days (when starts on day and finish on other day) on debate (start day and finish day).*

#### :pushpin: Related Issues
*Link your PR to an issue*
- Related to #30 

#### Testing
*Going to some one debate who have multiple days*

#### :clipboard: Checklist
:rotating_light: Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] :question: **CONSIDER** adding a unit test if your PR resolves an issue.
- [x] :heavy_check_mark: **DO** check open PR's to avoid duplicates.
- [x] :heavy_check_mark: **DO** keep pull requests small so they can be easily reviewed.
- [x] :heavy_check_mark: **DO** build locally before pushing.
- [ ] :heavy_check_mark: **DO** make sure tests pass.
- [x] :heavy_check_mark: **DO** make sure any new changes are documented in `docs/`.
- [x] :heavy_check_mark: **DO** add and modify seeds if necessary.
- [x] :heavy_check_mark: **DO** add CHANGELOG upgrade notes if required.
- [x] :heavy_check_mark: **DO** add to GraphQL API if there are new public fields.
- [x] :heavy_check_mark: **DO** add link to MetaDecidim if it's a new feature.
- [x] :x:**AVOID** breaking the continuous integration build.
- [x] :x:**AVOID** making significant changes to the overall architecture.

### :camera: Screenshots
**Before**
before
**After**
after

:hearts: Thank you!

### CODE REVIEW

**Review by davidbeig — COMMENTED** [review]
The test file has some lint problems. Could you solve them?

File: decidim-debates/app/views/decidim/debates/debates/show.html.erb [hunk 1]
```diff
@@ -42,14 +42,26 @@ edit_link(
...
+        <%= l debate.start_time, format: "%d" %> <span class="extra__month"> <%= l debate.start_time, format: "%B" %></span>
+        <div class="extra__time text-small"> <%= l debate.start_time, format: "%H:%M" %></span></div>
+      </div>
+      <div class="text-muted"><%= icon "arrow-thin-right" %></div>
+      <div>
+              <%= l debate.end_time, format: "%d" %> <span class="extra__month"> <%= l debate.end_time, format: "%B" %></span>
+              <div class="extra__time text-small"> <%= l debate.end_time, format: "%H:%M" %></span></div>
+      </div>
+          <% end %>
+  </div>
```
Comment (verarojman) [inline decidim-debates/app/views/decidim/debates/debates/show.html.erb]: Nice! It's working as it should :) To improve it, I would:

- Run `erblint` to correct the indentation (check the decidim docs if you're not sure how to do it)
- Extracting the logic into a [Rails helper](https://www.rubyguides.com/2020/01/rails-helpers/) would be super super nice 👌🏽  That way this file would be more readable and therefore easy to understand.

File: decidim-debates/app/views/decidim/debates/debates/show.html.erb [hunk 2]
```diff
@@ -42,14 +42,26 @@ edit_link(
...
+      <div>
+              <%= l debate.start_time, format: "%d" %> <span class="extra__month"> <%= l debate.start_time, format: "%B" %></span>
+        <div class="extra__time text-small"> <%= l debate.start_time, format: "%H:%M" %> - <%= l debate.end_time, format: "%H:%M" %></span></div>
+      </div>
+          <% else %>
+      <div>
+        <%= l debate.start_time, format: "%d" %> <span class="extra__month"> <%= l debate.start_time, format: "%B" %></span>
+        <div class="extra__time text-small"> <%= l debate.start_time, format: "%H:%M" %></span></div>
+      </div>
+      <div class="text-muted"><%= icon "arrow-thin-right" %></div>
```
Comment (verarojman) [inline decidim-debates/app/views/decidim/debates/debates/show.html.erb]: You can pass classes to the icon helper, like this:
```suggestion
      <%= icon "arrow-thin-right", class: "text-muted" %>
```

File: decidim-debates/spec/system/debates_dates.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe "Date comprober", versioning: true, type: :system do
```
Comment (verarojman) [inline decidim-debates/spec/system/debates_dates.rb]: "Date comprober" 🥇 👍🏽 Well, actually, it is a good practice for the name of the spec to match the file name!

```suggestion
describe "Debate dates", versioning: true, type: :system do
```

File: decidim-debates/spec/system/debates_dates.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+  before do
+    visit_component
+    click_link debate.title[I18n.locale.to_s], class: "card__link"
+  end
+
+  context "when shows the debate component" do
+    it "date same day" do
+	    Date.start_time.should be_between(Date.start_time.strftime("%h"),Date.end_time.strftime("%h"))
+    end
```
Comment (verarojman) [inline decidim-debates/spec/system/debates_dates.rb]: Where is this `Date` coming from?  😮  What is the output of this test?

Usually, in system specs for decidim (which uses Capybara), things are checked like this:

`expect(page).to have_content "foo bar"`

https://github.com/decidim/decidim/blob/release/0.23-stable/decidim-debates/spec/system/show_spec.rb

You should write specs for all the different cases: For different `start_time` and `end_time` combinations, check that the view has the content that you expect. More feedback coming when you do so! 🥳

**Review by verarojman — CHANGES_REQUESTED** [review]
Good work! I left some suggestions to improve the code, it seems it's not quite finished yet 🤷🏽‍♀️ please tell me if you have any doubts about the comments I left, I hope they're useful!

I'll review again once you add some more specs ;)

File: decidim-debates/spec/system/debates_dates_spec.rb [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      within ".extra__date" do
+        expect(page).to have_content(debate.start_time.strftime("%d"))
+        expect(page).to have_content(debate.start_time.strftime("%H:%M"))
+        expect(page).to have_content(debate.end_time.strftime("%H:%M"))
+      end
+    end
+
+    it "shows two hours" do
+      within ".extra__date" do
+        expect(debate.start_time.strftime("%H")).to be < debate.end_time.strftime("%H")
```
Comment (microstudi) [inline decidim-debates/spec/system/debates_dates_spec.rb:28]: You are check `debate` against `debate`? this kind of checks should be done in the model.
In here you need to check what's in the HTML view (the `page`). You might try to get the specific text for the element containing the displayed date

File: decidim-debates/app/views/decidim/debates/debates/show.html.erb [hunk 6]
```diff
@@ -44,16 +44,25 @@ edit_link(
...
+            <div class="extra__time text-small"> <%= l debate.start_time, format: "%H:%M" %></span></div>
+          </div>
+            <%= icon "arrow-thin-right", class: "extra__arrow" %>
+          <div>
+            <%= l debate.end_time, format: "%d" %> <span class="extra__month"> <%= l debate.end_time, format: "%B" %></span>
+            <div class="extra__time text-small"> <%= l debate.end_time, format: "%H:%M" %></span></div>
           </div>
         <% end %>
-
+      </div>
```
Comment (microstudi) [inline decidim-debates/app/views/decidim/debates/debates/show.html.erb]: Indentation is incorrect?

**Review by microstudi — CHANGES_REQUESTED** [review]
Tests are failing @cf17jeremy:
image

On the other hand, look at the comments, there are some things to improve!

**Pull request closed without merging**